tf_patches/small_constants_optimizer.diff

```diff
@@ -0,0 +1,41 @@
+diff --git a/tensorflow/core/common_runtime/eager/small_constants_optimizer.cc b/tensorflow/core/common_runtime/eager/small_constants_optimizer.cc
+index 0d6b36ea3d1..f7c439fb6aa 100644
+--- a/tensorflow/core/common_runtime/eager/small_constants_optimizer.cc
++++ b/tensorflow/core/common_runtime/eager/small_constants_optimizer.cc
+@@ -171,17 +171,17 @@ std::vector<FunctionDef> FoldBoolInputTensor(
+ 
+         // Promote the true branch when input_value is `true`.
+         if (attr_key == "then_branch") {
+-          if (input_value) node_def.mutable_attr()->emplace("f", attr_value);
++          if (input_value) node_def.mutable_attr()->insert({"f", attr_value});
+           continue;
+         }
+         // Promote the false branch when input_value is `false`.
+         if (attr_key == "else_branch") {
+-          if (!input_value) node_def.mutable_attr()->emplace("f", attr_value);
++          if (!input_value) node_def.mutable_attr()->insert({"f", attr_value});
+           continue;
+         }
+ 
+         // All other attributes should be copied over.
+-        node_def.mutable_attr()->emplace(attr_key, attr_value);
++        node_def.mutable_attr()->insert({attr_key, attr_value});
+       }
+     }
+ 
+@@ -213,13 +213,13 @@ std::vector<FunctionDef> FoldBoolInputTensor(
+   const_tensor->set_op("Const");
+   AttrValue dtype_value;
+   dtype_value.set_type(DT_BOOL);
+-  const_tensor->mutable_attr()->emplace("dtype", dtype_value);
++  const_tensor->mutable_attr()->insert({"dtype", dtype_value});
```

Comment:
are these already been fixed in head?

Reply:
It was fixed only at 05/03 after I talked to the original author while this PR updates to libtpu 05/01. To avoid the hassle (update again from 05/01 to 05/03), can we remove this diff in the next tf pin update?

Reply:
can we leave a comment on the top of this diff?

Reply:
done